sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java

```diff
@@ -162,6 +162,22 @@ private PollerFlux(Duration pollInterval,
         this.syncActivationOperation = cxt -> activationOperation.apply(cxt).block();
     }
 
+    /**
+     * Creates a poller flux instance that returns an error on subscription.
+     *
+     * @param ex The exception to be returned on subscription of this {@link PollerFlux}.
+     * @param <T> The type of poll response value.
+     * @param <U> The type of the final result of long running operation.
+     * @return A poller flux instance that returns an error without emitting any data.
+     *
+     * @see {@link Mono#error(Throwable)}
+     * @see {@link Flux#error(Throwable)}
+     */
+    public static <T, U> PollerFlux<T, U> error(Exception ex) {
+        return new PollerFlux<>(Duration.ofMillis(1L), context -> Mono.error(ex), context -> Mono.error(ex),
+            (context, response) -> Mono.error(ex), context -> Mono.error(ex));
+    }
```

Comment:
so this is the equivalent of `Mono.error` or `Flux.error` in `PollerFlux`.  The activation will never succeeded so all subscriptions are going to get the error from the activation [stage](https://github.com/Azure/azure-sdk-for-java/blob/51fdb024fad315a0c8273f0a9d84a5c3cf241b76/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java#L323). The other functions (finalResult, Poll) will never get called. lgtm.

Reply:
Yes, that's correct! This will return the error from activation function. Since poll, finalResult and cancel functions cannot be null, I just add the same error to all of them.

Reply:
was thinking a little more about this. So if we create the PollerFlux like below then parameter validations are delayed until consumer subscribe. 

```java
new PollerFlux<Foo, Void>(Duration.ofMillis(3),
    // The Activation Function
    new Function<PollingContext<Foo>, Mono<Foo>>() {
        @Override
        public Mono<Foo> apply(PollingContext<Foo> pollingContext) {
            Mono<Foo> initMono = startLro(start_params);
            return initMono;
        }
    },
    // The Poll Function
    new Function<PollingContext<Foo>, Mono<PollResponse<Foo>>>() {
        @Override
        public Mono<PollResponse<Foo>> apply(PollingContext<Foo> fooPollingContext) {
            Mono<PollResponse<Foo>> pollMono = doOnePoll(poll_params);
            return pollMono;
        }
    },
    // The Cancel Function
    new BiFunction<PollingContext<Foo>, PollResponse<Foo>, Mono<Foo>>() {
        @Override
        public Mono<Foo> apply(PollingContext<Foo> fooPollingContext,
                               PollResponse<Foo> fooPollResponse) {
            return Mono.error(new RuntimeException("not supported"));
        }
    },
    // The final result function
    new Function<PollingContext<Foo>, Mono<Void>>() {
        @Override
        public Mono<Void> apply(PollingContext<Foo> fooPollingContext) {
            return Mono.error(new RuntimeException("not supported"));
        }
    });
```


1. Any such delayed param validation error from the activation function will be bubbled up. 
2. But any such delayed param validation error from poll function will not bubbled up since the polling loop continue to poll on error.

The above behavior of two functions are same for both local param validation and server returned an error. The PollerFlux.error cannot be used in this pattern.
On the other hand, if we create the PollerFlux like below where we do eagerly do param validation then yes we can use PollerFlux.error to catch and convert to async error.

But this will not handle the error that poller function produce from server returned response.


```java

new PollerFlux<Foo, Void>(Duration.ofMillis(3),
    // The Activation Function
    getStartMonoFunction(start_params),
    // The Poll Function
    getPollMonoFunction(poll_params),
    // The Cancel Function
    new BiFunction<PollingContext<Foo>, PollResponse<Foo>, Mono<Foo>>() {
        @Override
        public Mono<Foo> apply(PollingContext<Foo> fooPollingContext,
                               PollResponse<Foo> fooPollResponse) {
            return Mono.error(new RuntimeException("not supported"));
        }
    },
    // The final result function
    new Function<PollingContext<Foo>, Mono<Void>>() {
        @Override
        public Mono<Void> apply(PollingContext<Foo> fooPollingContext) {
            return Mono.error(new RuntimeException("not supported"));
        }
    });

static Function<PollingContext<Foo>, Mono<Foo>> getStartMonoFunction(start_params) {
    // validation of "start_params" parameters and throwing of error outside reactor path
    return new Function<PollingContext<Foo>, Mono<Foo>>() {
        @Override
        public Mono<Foo> apply(PollingContext<Foo> pollingContext) {
            Mono<Foo> initMono = startLro();
            return initMono;
        }
    };
}

Function<PollingContext<Foo>, Mono<PollResponse<Foo>>> getPollMonoFunction(poll_params) {
     // validation of "poll_params" and throwing of error outside reactor path
    return new Function<PollingContext<Foo>, Mono<PollResponse<Foo>>>() {
        @Override
        public Mono<PollResponse<Foo>> apply(PollingContext<Foo> fooPollingContext) {
            Mono<PollResponse<Foo>> pollMono = doOnePoll();
            return pollMono;
        }
    };
}
```

Reply:
Yes, @anuchandy, you are right. The intention of creating this method is to do input validation in our public APIs. For e.g.

```java
public PollerFlux<Foo, Bar> beginPollingOperation(String data, Duration interval) {
    try {
        // All these input validations need to be done right here but will be surfaced to the user through proper error channel by returning PollerFlux.error()
         Objects.requireNonNull(data);
         Objects.requireNonNull(interval);
         if(interval.isZero()) {
               throw IllegalArgumentException("'interval' cannot be zero");
         }
         return new PollerFlux(.....);
    }
    catch (Exception ex) {
          return PollerFlux.error(ex);
    }
}
```